Data/models/DataController.swift

```diff
@@ -98,3 +98,16 @@ public class DataController: NSObject {
     }
 }
 
+extension NSManagedObjectContext {
+    public func saveContext () {
```

Comment:
We should probably only have one save context mechanism. Is there an issue with the other one?

Reply:
Well yeah we can remove this. I use this to merge contexts. Removing

Reply:
@danishjafri88 We shouldn't ever have to merge contexts.

Reply:
Not now perhaps, but there will be time where multiple contexts will come into play.

Reply:
We use lots of contexts, but generally they are just created for isolated instances, and main context auto-merges.

Reply:
Thats bad

Reply:
This is actually how `NSPersistentContainer` are setup to work. Contexts are lightweight and should be destroyed often, unless used to contain persistent data (e.g. `viewContext`). This was introduced in iOS10, and we follow it pretty closely. 

We can discuss this more elsewhere though. There are some context improvements we need to make but using many background contexts and an auto-merged main thread context are not some of them.